certbot/plugins/enhancement_test.py

```diff
@@ -0,0 +1,103 @@
+"""Tests for new style enhancements"""
+import unittest
+import mock
+
+from certbot import updater
+
+from certbot.plugins import enhancements
+
+import certbot.tests.util as test_util
+
+
+class EnhancementTest(test_util.ConfigTestCase):
+    """Tests for new style enhancements in certbot.plugins.enhancements"""
+
+    def setUp(self):
+        super(EnhancementTest, self).setUp()
+        self.mockinstaller = test_util.MockInstallerAutoHSTS()
+
+    @mock.patch('certbot.plugins.selection.choose_configurator_plugins')
+    @test_util.patch_get_utility()
+    def test_enhancement_updates(self, _, mock_select):
+        mock_select.return_value = (self.mockinstaller, None)
+        updater.run_generic_updaters(self.config, mock.MagicMock(), None)
```

Comment:
Sorry to keep nitpicking this, but I think tests that are calling into other modules like `updater` should be deleted or moved to that module. I think we should just be testing the functions in `enhancements` here. The tests should be very simple.

Reply:
This makes a lot of sense, and the tests have now been refactored as such.